docs/state_encryption.md

```diff
@@ -0,0 +1,333 @@
+
+Users will be able to specify their encryption configuration both in code and via environment variables. Both configurations are equivalent and will be merged at execution time. For more details, see the [environment configuration](#environment-configuration) section below.
+
+When a user wants to enable encryption, they must specify the following block:
+
+```hcl2
+terraform {
+  encryption {
+    // Encryption options
+  }
+}
+```
+
+The mere presence of the `encryption` block alone should not enable encryption because the user should explicitly specify what key and method to use. The implementation should warn the user if the encryption block is present but has no configuration.
+
+The encryption relies on an encryption key, or a composite encryption key, which the user can provide directly or via a key management system. The user must provide at least one `key_provider` block with the settings described below. These key providers serve the purpose of creating or providing the encryption key. For example, the user could hard-code a static key named foo:
+
+```hcl2
+terraform {
+  encryption {
+    key_provider "static" "foo" {
+      key = "6f6f706830656f67686f6834616872756f3751756165686565796f6f72653169"
+    }
+  }
+}
+```
+
+**Note:** the user is responsible for keeping this key safe and follow disaster recovery best practices. OpenTofu is a read-only consumer for the key the user provided and will not perform tasks on the key management system like key rotation.
+
+The user also has to specify at least one encryption method referencing the key provider. This encryption method determines how the encryption takes place. It is the user's responsibility to make sure that they provided a key that is suitable for the encryption method.
+
+```hcl2
+terraform {
+  encryption {
+    //...
+    method "aes_gcm" "bar" {
+      key_provider = key_provider.static.foo
+    }
+  }
+}
+```
+
+Finally, the user must reference the method for use in their state file, plan file, etc. This enables creating a different configuration for different purposes.
+
+```hcl2
+terraform {
+  encryption {
+    //...
+    statefile {
+      method = method.aes_gcm.abc
+    }
+    planfile {
+      method = method.aes_gcm.cde
+    }
+    backend {
+      method = method.pbkdf2.efg
+    }
+    remote_data_sources {
+      default {
+        method = method.aes_gcm.ghi
+      }
+      remote_data_source "some_module.remote_data_source.foo" {
+        method = method.aes_gcm.ijk
+      }
+    }
+  }
+}
+```
+
+To facilitate key and method rollover, the user can specify a fallback configuration for state and plan decryption. When the user specifies a `fallback` block, `tofu` will first attempt to decrypt any state or plan it reads with the primary method and then fall back to the `fallback` method. When `tofu` writes a state or plan, it will not use the `fallback` method and always writes with the primary method.
```

Comment:
I think it's worth noting here that if there *is no* primary, and there *is a* fallback, then tofu *will not* encrypt anything (after successfully decrypting it on read).

Reply:
Correct, I'll update the document to reflect this.